docs/source/onedal/algorithms/decomposition/pca.rst

```diff
@@ -33,68 +33,89 @@ Mathematical formulation
+
+The eigenvector associated with the :math:`k`-th largest eigenvalue of the covariance (or correlation) matrix 
+is also the :math:`k`-th principal component of the training data set.  Based on this principle,
+the principal components can be computed in three steps:
 
-#. Computation of the covariance matrix
-#. Computation of the eigenvectors and eigenvalues
-#. Formation of the matrices storing the results
+#. Computation of the covariance (or correlation) matrix
+#. Computation of the eigenvectors and eigenvalues of the covariance (or correlation) matrix
+#. Processing (sorting and storing) the results
 
-Covariance matrix computation is performed in the following way:
 
-#. Compute the vector-column of sums :math:`s_i = \sum_{j=1}^n x_{i,j}, \quad 1 \leq i \leq p`.
+Covariance matrix can be computed in the following way:
 
-#. Compute the cross-product :math:`P = X^TX - s^Ts`.
+#. Compute the column means :math:`M = (M_j)`,:math:`j = \overline{1,p}`, :math:`M_j = \frac{1}{n}\sum _{i} X_{ij}`.
+#. Compute the sample covariance matrix :math:`S = (S_{ij})`, :math:`i=\overline{1,p}`, :math:`j=\overline{1,p}`, :math:`S_{ij} = \frac{1}{n-1}\sum_{k=1}^{n}(X_{ki} - M_i)(X_{kj}-M_j)`.
 
-#. Compute the covariance matrix :math:`\Sigma = \frac{1}{n - 1} P`.
+Corelation matrix can be computed from covariance matrix in the following way:
 
-To compute eigenvalues :math:`\lambda_i` and eigenvectors :math:`\upsilon_i`, the implementer can choose an arbitrary
+#. Compute the correlation matrix :math:`C = C_{ij}`, :math:`i=\overline{1,p}`, :math:`j=\overline{1,p}`,:math:`C_{ij} = \frac{S_{ij}}{\sqrt{S_{ii}\cdot S_{jj}}}`.
+
+
+The eigenvalues :math:`\lambda_k` and eigenvectors :math:`\upsilon_k` can be computed by an arbitrary
 method such as [Ping14]_.
 
-The final step is to sort the set of pairs :math:`(\lambda_i, \upsilon_i)` in
-the descending order by :math:`\lambda_i` and form the resulting matrix :math:`T
-= (\upsilon_{i,1}, \cdots, \upsilon_{i,r}), \quad 1 \leq i \leq p`.
-Additionally, the means and variances of the initial dataset are returned.
+In the final step, the eigenvalues (:math:`\lambda_k`) are sorted in descending order 
+to determine the order of the principal components. Each principal component is 
+stored as a row of the final resulting matrix, :math:`T = (\upsilon_{1j}, \cdots, \upsilon_{rj}), \quad 1 \leq j \leq p`.
+Additionally, the means and variances of the input dataset are returned.
 
 .. _pca_t_math_svd:
 
 Training method: *SVD*
 ~~~~~~~~~~~~~~~~~~~~~~
 
-This method uses singular value decomposition of the dataset to compute its
-principal components. The method relies on the following steps:
+The singular value decomposition (SVD) is a matrix factorization technique that
+decomposes an observation matrix :math:`X_{n \times p}` into three matrices as :math:`X = U\SigmaV^*`.
+Here, the columns of :math:`U` are the left-singular vectors, the
```

Comment:
I'd recommend making it a bullet list. 
Here:
* The columns of :math:`U` are the left-singular vectors.
* The columns of :math:`V` are the right-singular vectors.
* :math:`V^*` is the conjugate transpose of the matrix :math:`V`.
* The diagonal entries of :math:`\Sigma` are the singular values (:math:`\sigma`) of :math:`X`.